lib/cequel/schema/table.rb

```diff
@@ -157,11 +157,26 @@ def data_columns
       # @return [void]
       #
       def add_data_column(name, type, options = {})
-        options = {index: options} unless options.is_a?(Hash)
-        index_name = options[:index]
-        index_name = :"#{@name}_#{name}_idx" if index_name == true
-        DataColumn.new(name, type(type), index_name)
-          .tap { |column| data_columns_hash[name] << add_column(column) }
+        if valid_data_column?(name)
+          options = {index: options} unless options.is_a?(Hash)
+          index_name = options[:index]
+          index_name = :"#{@name}_#{name}_idx" if index_name == true
+          DataColumn.new(name, type(type), index_name)
+            .tap do |column|
+              data_columns_hash[name] = add_column(column)
+            end
+        else
+          # return the data column like we would if we built it
+          data_columns_hash[name]
+        end
```

Comment:
Yeah, I thought about that a lot, and how that would work in the case of migrations. But, then, I decided that made my head hurt.. so I did it this way. I think though that it's a good thought, and I definitely thought it as well. I'll see what I can do on my next pull request.